tests/test_consumed.py

```diff
@@ -0,0 +1,169 @@
+#-----------------------------------------------------------------------------#
+#   test_consumed.py                                                          #
+#                                                                             #
+#   Copyright (c) 2017-2018, Rajiv Bakulesh Shah, original author.            #
+#   All rights reserved.                                                      #
+#-----------------------------------------------------------------------------#
+
+
+import unittest
+
+from bloom import RecentlyConsumed
+
+
+class RecentlyConsumedTests(unittest.TestCase):
+    def setUp(self):
+        super(RecentlyConsumedTests, self).setUp()
+
+        # Wait for each Memcache command to complete before moving on to the
+        # next line of code.  This makes our tests run deterministically.
+        RecentlyConsumed._NOREPLY = False
+
+        self.consumed = RecentlyConsumed(key='consumed:rajiv', maxlen=10)
+
+    def tearDown(self):
+        self.consumed.clear()
+        RecentlyConsumed._NOREPLY = True
+        super(RecentlyConsumedTests, self).tearDown()
+
+    def test_maxlen_is_not_writable(self):
+        with self.assertRaises(AttributeError):
+            self.consumed.maxlen = 100
+
+    def test_memcached_list_does_not_exceed_maxlen(self):
+        self.consumed.extend(('t3_1', 't3_2', 't3_3', 't3_4', 't3_5', 't3_6'))
+        with self.assertRaises(IndexError):
+            RecentlyConsumed(key='consumed:rajiv', maxlen=5)
```

Comment:
This is interesting. So if we decide at some point to reduce `maxlen`, will we need to clear memcached first to avoid hitting the error? Ideally we should be able to deploy a change like that without losing the data.